classes/metadata/MetadataProperty.php

```diff
@@ -423,8 +423,16 @@ public function isValid($value, $locale = null)
 
                             if (is_string($value)) {
                                 // Try to translate the string value into a controlled vocab entry
-                                $controlledVocabEntryDao = DAORegistry::getDAO('ControlledVocabEntryDAO'); /** @var ControlledVocabEntryDAO $controlledVocabEntryDao */
-                                if (!is_null($controlledVocabEntryDao->getBySetting($value, $symbolic, $assocType, $assocId, 'name', $locale))) {
+                                $entry = ControlledVocabEntry::query()
+                                    ->whereHas(
+                                        'controlledVocab',
+                                        fn ($query) => $query->withSymbolic($symbolic)->withAssoc($assocType, $assocId)
+                                    )
+                                    ->withLocale($locale)
+                                    ->withSetting($symbolic, $value)
```

Comment:
here previously setting name passed as `name` but is that correct as for controlled vocabs we have specific settings ?